www/js/profile/profile.js

```diff
@@ -1,17 +1,29 @@
 'use strict';
 
-angular.module('main').controller('ProfileCtrl', ['$scope', '$state', '$rootScope', 'AuthService', function ($scope, $state, $rootScope, AuthService) {
-    AuthService.getLoggedInUser()
-    .then(function(user){
-      $scope.profile = user;
-    })
-    .catch(function(err){
-      console.log(err);
-    });
-
+angular.module('main').controller('ProfileCtrl', ['$scope', '$state', '$rootScope', 'AuthService', '$cordovaSQLite', function ($scope, $state, $rootScope, AuthService, $cordovaSQLite) {
+    var unbind, fireBaseObj;
     $scope.logOut = function(){
       AuthService.logout();
-      $state.go('tab.login');
+      $state.go('tab.login');      
+      $scope.loggedIn = false;
+      if (unbind) unbind();
+      fireBaseObj.$destroy();
     };
+
+
+    $scope.$on("$ionicView.enter", function () {
```

Comment:
Huh cool. This solves what we were talking about right—that controller code was not reloading? Have you confirmed this will only run its callback for the current state/view? That it does not execute for *any/every* "ionic view enter"?